### PR TITLE
tree: type tests tree public API

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

With the recent type test fixes, tree's public API can be type tested.

This enables those type tests.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/dds/tree/.eslintrc.cjs [hunk 1]
```diff
@@ -70,6 +70,12 @@ module.exports = {
 				"@typescript-eslint/explicit-function-return-type": "off",
 			},
 		},
+		{
```
Comment (Abe27342) [inline packages/dds/tree/.eslintrc.cjs:73]: seems worth augmenting the note on where this is enabled or duplicating the comment here

```ts
// TODO: Remove this override once this config has been updated to extend at least the "recommended" base config.
```

File: packages/dds/tree/package.json [hunk 2]
```diff
@@ -149,6 +149,7 @@
 		"@fluidframework/eslint-config-fluid": "^5.3.0",
 		"@fluidframework/test-runtime-utils": "workspace:~",
 		"@fluidframework/test-utils": "workspace:~",
+		"@fluidframework/tree-previous": "npm:@fluidframework/tree@2.0.0",
```
Comment (Abe27342) [inline packages/dds/tree/package.json:152]: I'm not up-to-date on recent changes in typetests, but we probably want this to align with other packages' "previous" install. Looks like client packages are generally still installing 2.0.0-rc.5.0.0 as previous, hence larger-than-expected diff to package-lock.

Comment (CraigMacomber) [reply packages/dds/tree/package.json:152]: Looks like the other packages need to be updated, as part of our release process is to update the type tests to target the most recent minor version (2.0). I'll land that before landing this.

File: packages/dds/tree/.eslintrc.cjs (hunk 1 above)
Comment (CraigMacomber) [reply packages/dds/tree/.eslintrc.cjs:73]:
```suggestion
		// TODO: Remove this override once this config has been updated to extend at least the "recommended" base config.
		{
```

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**